test/ably/rest/restchannelpublish_test.py

```
@@ -402,26 +401,31 @@ async def test_interoperability(self):
+                await assert_waiter(check_data)
 
                 # 2)
-                await channel.publish(messages=[Message(data=msg_data, encoding=encoding)])
-                # temporary added delay, we need to investigate why messages don't appear immediately
-                await asyncio.sleep(1)
-                history = await channel.history()
-                message = history.items[0]
-                assert message.data == expected_value
-                assert type(message.data) == type_mapping[expected_type]
+                response = await channel.publish(messages=[Message(data=msg_data, encoding=encoding)])
+                assert response.status_code == 201
```

Ruff (0.12.2)

Use of `assert` detected

(S101)